src/test/java/seedu/address/model/person/AppointmentTest.java

```diff
@@ -3,153 +3,112 @@
-        Doctor validDoctor = DEREK;
-        Patient validPatient = AMY;
-        LocalDateTime invalidAppointmentTime = null;
-        assertThrows(NullPointerException.class, () -> new Appointment(validDoctor, validPatient,
-                invalidAppointmentTime));
+    public void constructor_nullAppointmentTime_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Appointment(DEREK, AMY, null));
     }
 
     @Test
-    public void secondConstructor_invalidDoctor_throwsIllegalArgumentException() {
-        Doctor invalidDoctor = null;
-        Patient validPatient = AMY;
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
-        LocalDateTime validAppointmentTime = LocalDateTime.parse("2022-02-14 13:30:00", formatter);
-        assertThrows(NullPointerException.class, () -> new Appointment(invalidDoctor, validPatient,
-                validAppointmentTime, "Follow-Up"));
+    public void secondConstructor_nullDoctor_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Appointment(null, AMY,
+                VALID_DATE_1, "Follow-Up"));
     }
     @Test
-    public void secondConstructorr_invalidPatient_throwsIllegalArgumentException() {
-        Doctor validDoctor = DEREK;
-        Patient invalidPatient = null;
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
-        LocalDateTime validAppointmentTime = LocalDateTime.parse("2022-02-14 13:30:00", formatter);
-        assertThrows(NullPointerException.class, () -> new Appointment(validDoctor, invalidPatient,
-                validAppointmentTime, "Follow-Up"));
+    public void secondConstructorr_nullPatient_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Appointment(DEREK, null,
+                VALID_DATE_1, "Follow-Up"));
     }
 
     @Test
-    public void secondConstructor_invalidAppointmentTime_throwsIllegalArgumentException() {
-        Doctor validDoctor = DEREK;
-        Patient validPatient = AMY;
-        LocalDateTime invalidAppointmentTime = null;
-        assertThrows(NullPointerException.class, () -> new Appointment(validDoctor, validPatient,
-                invalidAppointmentTime, "Follow-Up"));
+    public void secondConstructor_nullAppointmentTime_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Appointment(DEREK, AMY,
+                null, "Follow-Up"));
     }
     @Test
-    public void constructor_nullStatus() {
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
-        LocalDateTime appointmentTime = LocalDateTime.parse("2022-02-14 13:30:00", formatter);
-        assertThrows(NullPointerException.class, () -> new Appointment(DEREK, AMY, appointmentTime, null));
+    public void constructor_nullStatus() { // should this throw a nullPointerException?
```

Comment:
I think it makes sense to throw null pointer exception here, because if there is no status given, appointment should be initialised with 3 parameters instead, which will call the other constructor instead of this.